patchs/patch

```diff
@@ -0,0 +1,72 @@
+diff --git a/internal/k8s/main.go b/internal/k8s/main.go
```

Comment:
whats going on here? is this a patch we've always carried in our version of ingress-ngix or is this a ne bit of patched code?

Reply:
This patch is for our customized changes. Following changes are included in this patch
- It allows us to use rke node ip annotations(`rke.cattle.io/internal-ip` and `rke.cattle.io/external-ip`) when external-ip is not set in the node.
- The nginx ingress is using labels of itself to find their peer instances. But Rancher will add unique label to pods if the ingress-nginx daemonset get changed in Rancher UI. So to prevent this case, I am using the selector of ingress-nginx daemonset to find peer instances instead.
- We reuse the test process of ingress-nginx but we can't do e2e test in dapper. So I skip the test for now.

I make it as a patch to let it easier to rebase to other ingress version. Only adding changes in the rebase PR and the ci process will tell the patch is validated or not.

Reply:
I already remove the patch and using commit again.